pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -270,15 +326,82 @@ func (r *Reconciler) validateVpcSubnetsBySubnetCR(ctx context.Context, namespace
 		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByChildSubnet(subnets[0])
 		if len(bms) > 0 {
 			dependency := r.SubnetBindingService.GetSubnetConnectionBindingMapCRName(bms[0])
-			return nil, &errorWithRetry{
+			return nil, subnetCR, &errorWithRetry{
 				message: fmt.Sprintf("Target Subnet CR %s is associated by %s", name, dependency),
 				error:   fmt.Errorf("target Subnet %s is already associated by SubnetConnectionBindingMap %s", name, dependency),
 				retry:   true,
 			}
 		}
 	}
+	return subnets, subnetCR, nil
+}
 
-	return subnets, nil
+func (r *Reconciler) getSharedSubnetFromNSX(subnet *v1alpha1.Subnet) (*model.VpcSubnet, *errorWithRetry) {
+	anno := subnet.GetAnnotations()
+	associatedResource, ok := anno[servicecommon.AnnotationAssociatedResource]
+	if !ok {
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("No associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name),
+			error:   fmt.Errorf("no associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name),
+			retry:   false,
+		}
+	}
+	// associatedResource has the format projectID:vpcID:subnetID or :vpcID:subnetID for default project
+	parts := strings.Split(associatedResource, ":")
+	if parts[0] == "" {
+		nc, err := r.VPCService.GetDefaultNetworkConfig()
+		if err != nil {
+			return nil, &errorWithRetry{
+				message: fmt.Sprintf("Failed to get default NetworkConfig, error: %v", err),
+				error:   err,
+				retry:   true,
```

Comment:
Could you add comments to explain what error is retrieable and what is not?

Reply:
Added to subnet uts as the function is moved there, thanks!